lib/cipherstash/index.rb

```diff
@@ -35,6 +35,19 @@ def subclass_from_kind(kind)
         end
       end
 
+      def invalid_unique_constraint?(unique, kind)
+        [
+          "match",
+          "ore-match",
+          "filter-match",
+          "field-dynamic-match",
+          "field-dynamic-filter-match",
+          "field-dynamic-ore-match",
+          "dynamic-match",
+          "dynamic-filter-match",
+          "dynamic-ore-match"].include?(kind) && unique
```

Comment:
What happens if a value other than a boolean is given for `unique`?

Reply:
Unique would evaluate to true, unless nil or false is passed.

Bu this has been refactored out to the above suggestion now, so the method is on each subclass. I like that this check get's straight to the point `base_settings.key?("unique")`, if there is a unique key and the index doesn't support uniqueness, raise.